### PR TITLE
add links checkers the crawls the site for 404s

### PR DESCRIPTION
**Pull request by vanessayuenn**
Results compile to #1022. Right now the command needs to be run manually, but after we've fixed the known 404s, we can look into adding this as the last step of the build process.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -14,7 +14,8 @@
     "release": "node script/release",
     "dev": "cross-env NODE_PATH=. nodemon -e js,scss,yml,md server.js",
     "prepack": "check-for-leaks",
-    "prepush": "check-for-leaks"
+    "prepush": "check-for-leaks",
+    "linkschecker": "NODE_PATH=. NODE_ENV=test node scripts/links-checker.js"
```
Comment (zeke) [inline package.json]: Why do `NODE_PATH` and `NODE_ENV` need to be set?

File: scripts/links-checker.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+const app = require('../server.js')
+const port = Number(process.env.PORT) || argv.p || argv.port || 5000
+const host = process.env.HOST || `http://localhost:${port}`
+
+const crawler = require('simplecrawler')(host)
+crawler.parseScriptTags = false
+
+crawler.addFetchCondition((queueItem, referrerQueueItem, cb) => {
+  cb(null, !queueItem.path.match(/\/images|styles\//gi))
+  // cb(null, queueItem.path.match(/\/docs\//gi))
```
Comment (zeke) [inline scripts/links-checker.js]: This comment can be removed.

File: scripts/links-checker.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  const $ = cheerio.load(buffer.toString('utf8'))
+  return $('a[href]').get().map((elem) => $(elem).attr('href'))
+}
+
+crawler.on('fetch404', (queueItem, response) => {
+  console.warn(`ERROR(404) fetching ${queueItem.path} (referrer: ${queueItem.referrer.replace(host, '')})`)
+})
+
+crawler.on('complete', () => {
+  console.log('fin')
```
Comment (zeke) [inline scripts/links-checker.js]: 👌

File: package.json (hunk 1 above)
Comment (vanessayuenn) [reply package.json]: it's so the server would start properly, and without the browsersync stuff.

**Pull request merged**